content/blog/2017-09-08-dom-attributes-in-react-16.md

```diff
@@ -43,127 +43,127 @@ title: 'title',
 // ...
 ```
 
-This had two downsides:
+Cela avait deux inconvénients:
 
-* You could not [pass a custom attribute](https://github.com/facebook/react/issues/140). This is useful for supplying browser-specific non-standard attributes, trying new DOM APIs, and integrating with opinionated third-party libraries.
+* Vous ne pouviez pas [passer un attribut personnalisé](https://github.com/facebook/react/issues/140). Cela est utile pour fournir des attributs non standard spécifiques au navigateur, essayer de nouvelles API DOM et s'intégrer à des bibliothèques tierces avisées.
 
-* The attribute list kept growing over time, but most React canonical attribute names are already valid in the DOM. Removing most of the whitelist helped us reduce the bundle size a little bit.
+* La liste d'attributs n'a cessé de croître au fil du temps, mais la plupart des noms d'attributs canoniques React sont déjà valides dans le DOM. La suppression de la majorité de liste blanche nous a aidé à diminuer un peu la taille du paquet.
 
-With the new approach, both of these problems are solved. With React 16, you can now pass custom attributes to all HTML and SVG elements, and React doesn't have to include the whole attribute whitelist in the production version.
+Avec la nouvelle approche, ces deux problèmes sont résolus. Avec React 16, vous pouvez désormais transmettre des attributs personnalisés à tous les éléments HTML et SVG, et React n'a pas à inclure de liste blanche d'attributs dans sa version de production.
 
-**Note that you should still use the canonical React naming for known attributes:**
+** Notez que vous devez toujours utiliser la dénomination canonique React pour les attributs connus: **
 
 ```js
-// Yes, please
+// Oui, s'il vous plaît
 <div tabIndex="-1" />
 
-// Warning: Invalid DOM property `tabindex`. Did you mean `tabIndex`?
+// Attention: propriété du DOM invalide `tabindex`. Vous voulez certainement dire `tabIndex`?
 <div tabindex="-1" />
 ```
 
-In other words, the way you use DOM components in React hasn't changed, but now you have some new capabilities.
+En d'autres termes, la façon dont vous utilisez les composants du DOM dans React n'a pas changé, mais vous disposez désormais de nouvelles fonctionnalités.
 
-## Should I Keep Data in Custom Attributes? {#should-i-keep-data-in-custom-attributes}
+## Dois-je conserver les données dans des attributs personnalisés ? {#dois-je-conserver-les-donnees-dans-des-attributs-personnalises}
 
-No. We don't encourage you to keep data in DOM attributes. Even if you have to, `data-` attributes are probably a better approach, but in most cases data should be kept in React component state or external stores.
+Non. Nous ne vous encourageons pas à conserver les données dans les attributs DOM. Même si vous le devez, les attributs `data-` sont probablement une meilleure approche, mais dans la plupart des cas, les données doivent être conservées dans l'état du composant React ou dans des collections externes.
```

Comment:
```suggestion
Non. Nous vous déconseillons de stocker des données dans les attributs DOM. Même si vous en aviez besoin, les attributs `data-` constituent probablement une meilleure approche, mais dans la plupart des cas, les données devraient être conservées dans l'état du composant React ou dans des *stores* externes.
```

- Should ≠ Must, donc « devrait » plutôt que « doit »
- En français on ne dira pas «  ne pas encourager » mais « déconseiller » (et pas non plus « décourager »)
- “Store” a un sens métier particulier ici, sans VF bien établie, du coup on le garde tel quel mais italicisé.